src/main/scala/ylabs/messaging/Client.scala

```diff
@@ -22,6 +22,9 @@ object Client {
 
   case class User(value: String) extends AnyVal
   case class Password(value: String) extends AnyVal
+  case class Domain(value: String) extends  AnyVal
```

Comment:
hmm, looks like you went a bit overboard with all these case classes. How about we only keep `case class User(name: String, domain: Option[String])?